src/metabase/query_processor/middleware/process_userland_query.clj

```diff
@@ -32,23 +37,85 @@
 ;;; |                                              Save Query Execution                                              |
 ;;; +----------------------------------------------------------------------------------------------------------------+
 
+;; Should be somewhere in metabase.lib
+(defn- remove-lib-uuids
+  "Two queries should be the same even if they have different :lib/uuids, because they might have both been converted
+  from the same legacy query."
+  [x]
+  (walk/postwalk
+   (fn [x]
+     (if (map? x)
+       (dissoc x :lib/uuid)
+       x))
+   x))
+
+(defn- pmbql->field-usages
```

Comment:
to QP people: do you think I should break this down and make finding aggregation, breakout, filters, expressions an mlv2 function? if yes, what'd be an useful API?